builtin/logical/database/path_config_connection.go

```diff
@@ -55,16 +58,8 @@ func (b *databaseBackend) pathConnectionReset() framework.OperationFunc {
 			return logical.ErrorResponse(respErrEmptyName), nil
 		}
 
-		// Grab the mutex lock
-		b.Lock()
-		defer b.Unlock()
-
 		// Close plugin and delete the entry in the connections cache.
-		b.clearConnection(name)
-
-		// Execute plugin again, we don't need the object so throw away.
```

Comment:
Why did we remove this?

Reply:
They have separate locks so it will get created at next call.

Reply:
I guess the nice thing about this was it would return an error if there was an issue starting the plugin, I wonder if we want to continue to do that?

Reply:
I added it back.  This is not done under the same lock but I think it is ok, we just want to verify we can get a new plugin instance and if something already created one, that should be ok.